### ISSUE TITLE
[CRASH] Pegged CPU causing Segmentation Faults

### ISSUE THREAD

**Issue by Matt-Hapner** [OpenSIPS/opensips#2452]
**OpenSIPS version you are running**

```
opensips -V
version: opensips 2.4.9 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 3bed2f312
main.c compiled on 16:01:30 Mar 25 2021 with gcc 7
```

**Crash Core Dump**

Two different core dump traces off of two instances:
https://pastebin.com/Z8BhFjGS
https://pastebin.com/NXEQ1rMY

**Describe the traffic that generated the bug**

OpenSIPS was processing calls on multiple different instances where the CPU was running at a higher than normal percentage. This was, in part, due to an increase in call volume, but also due to another process that was running alongside OpenSIPS that was hogging resources (CPU time, memory, etc.). OpenSIPS did not properly recognize the resource starvation and crashed with segmentation faults in the SSL libraries.

**To Reproduce**

No steps to note, other than that the CPU was pegged on the instances.

**Relevant System Logs**

```
Mar 29 19:29:16 [8245] ERROR:tm:msg_send:   send() to 34.192.178.111:56312 for proto tls/3 failed
Mar 29 19:29:16 [8245]   ERROR:core:add_write_chunk: We have reached the limit of max async postponed   chunks
Mar 29 19:29:16 [8245]   ERROR:core:proto_tcp_send: Failed to add another write chunk to   0x7f24a9163b68
```

**OS/environment information**
 - Operating System: amazon-linux
 - OpenSIPS installation: 
 - other relevant information:

**Additional context**

**Comment by bogdan-iancu** [comment]
@Matt-Hapner , your version a bit old (looks like from December)  and we had Feb this year some fixes in the SSL related code about potential races when accessing the SSL contexts. The high load probably made the race more likely.
Try to get the latest 2.4 from GIT or via nightly-builds (if you use packages).

**Comment by bcnewlin** [comment]
@bogdan-iancu We attempted to upgrade to the latest version but there is a statistics bug causing test failures. I've opened #2460.

**Comment by bogdan-iancu** [comment]
Not sure if possible, but just ignore that statistics, as it is buggy - most probably it will be removed.

**Comment by bogdan-iancu** [comment]
@bcnewlin , I think you have a green light now for upgrade, thank to 4b076defcfbfa41c3e83ba78285f0340c38162a0, right ?

**Comment by bcnewlin** [comment]
@bogdan-iancu Yes, we have that fix, but unfortunately we are still blocked on the upgrade as the format of one of the CDR fields has been changed in a commit since the one we are on. We are working to accommodate the new format, but this is an unexpected breaking change for minor version fix. We are not sure which commit made the change specifically. Also, we use the text DB module and we're not sure if the change is specific to that module.

The change is that the <field> ACC parameter has changed from an int value to a float value. We have not opened a ticket for this against OpenSIPS as we plan to accommodate the change on our end but we can if desired.

**Comment by bogdan-iancu** [comment]
@bcnewlin , I do not recall such a change between 2.4.9 and 2.4.10 - could you provide more info on the issue you have ?

**Comment by bcnewlin** [comment]
@bogdan-iancu Let me get back with the team doing the testing and I will get you more details.

**Comment by bcnewlin** [comment]
@bogdan-iancu Apologies, after further investigation we believe the change was on our end. We are now running the latest OpenSIPS commit.

```
$ opensips -V
version: opensips 2.4.9 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 078887fef
main.c compiled on 15:07:57 Apr 22 2021 with gcc 7
```

We will continue to monitor, but we have not seen a recurrence of the issue even on our older version and we have not been able to reproduce.

**Comment by bogdan-iancu** [comment]
@bcnewlin , if you have no new data here, feel free to close. IMO, this should have been fixed with the upgrade.

**Comment by bcnewlin** [comment]
@bogdan-iancu Yes, I am fine to close.
